pkgs/tools/networking/wireguard-go/deps.nix

```diff
@@ -43,6 +57,1748 @@
       url = "https://go.googlesource.com/tools";
       rev = "90fa682c2a6e";
       sha256 = "03ic2xsy51jw9749wl7gszdbz99iijbd2bckgygl6cm9w5m364ak";
+      moduleDir = "";
+    };
+  }
+    {
+    goPackagePath = "bazil.org/fuse";
```

Comment:
After version update source didn't compile due to missing packages
errors like
```
go/src/golang.zx2c4.com/wireguard/tun/netstack/tun.go:24:2: cannot find package "gvisor.dev/gvisor/pkg/tcpip" in any of:

        /nix/store/sppqwgfpqhjl1lk90gxfdyx2jqxbbfwk-go-1.16.7/share/go/src/gvisor.dev/gvisor/pkg/tcpip (from $GOROOT)
        /build/go/src/gvisor.dev/gvisor/pkg/tcpip (from $GOPATH)
```
Inspecting $src I found that there is additional dependencies file : $src/tun/netstack/go.mod
I've executed "vgo2nix" in the $src/tun/netstack dir and added content of generated deps.nix into root deps.nix

Is that common for project to have separate levels declaring different dependencies?
Would the correct way to solve this problem is to offer modification to "vgo2nix" that would check project filestricture recursively, searching for other do.mod files?

Reply:
The netstack go.mod is source-only, and is not part of the wireguard-go binary at all. As such, it shouldn't have its deps listed in deps.nix.

Reply:
I had that suspicion, but didn't know enough to confirm that without `tun/netstack` the built program would work 

And both `buildGoPackage` and `buildGoModule` attempted to build that module and failed
`..package` failed if no dependencies were provided
`..module` failed with `main module (golang.zx2c4.com/wireguard) does not contain package golang.zx2c4.com/wireguard/tun/netstack`

For educational purpose could you tell me what did you to figure with sertainty out that `tun/netstack` is not a part of "wireguard-go" binary?

And what would you think about possible solution to both `buildGo` functions attempting to build it:
- add patch that would remove the `/tun/netstack` directory
- use `buldGoModule`
presence of the `/tun/netstack` seemed to be the only problem during the `buildGoModule` buildPhase

Reply:
```
zx2c4@thinkpad ~/Projects/wireguard-go $ grep -r tun/netstack
tun/netstack/go.mod:module golang.zx2c4.com/wireguard/tun/netstack
tun/netstack/examples/http_client.go:   "golang.zx2c4.com/wireguard/tun/netstack"
tun/netstack/examples/http_server.go:   "golang.zx2c4.com/wireguard/tun/netstack"
```

It's not imported by anything at all.

I'm not sure what logic `buildGoModule` is using, but `go build` doesn't include it or need it:

```
zx2c4@thinkpad ~/Projects/wireguard-go $ go build -v -o wireguard-go -a 2>&1 | grep netstack
zx2c4@thinkpad ~/Projects/wireguard-go $ 
```